### PR TITLE
Allow to use max timeout for get status and allow to specify the max timeout

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Instead of retrying to fetch the status with multiple different timeouts, we should fetch the status directly with the maximum timeout. This reduced the amount of retries if the cluster is not responsive. I create a follow up for a better retry mechanism.

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

## Discussion

-

## Testing

e2e tests will be running.

## Documentation

-

## Follow-up

-

### CODE REVIEW

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [20246c7](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/20246c7f62e2b938413f897d3f03be49e893de3d)
> * Duration 3:19:43
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/7eb7c547-6095-4884-a3f5-a5f0ec9123fa.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/7eb7c547-6095-4884-a3f5-a5f0ec9123fa/workspace.zip) zip file of the working directory (available for 30 days)

```text

File: fdbclient/admin_client.go [hunk 1]
```diff
@@ -537,7 +502,7 @@ func cleanConnectionStringOutput(input string) string {
 func (client *cliAdminClient) GetConnectionString() (string, error) {
 	// This will call directly the database and fetch the connection string
 	// from the system key space.
-	outputBytes, err := getConnectionStringFromDB(client.fdbLibClient)
+	outputBytes, err := getConnectionStringFromDB(client.fdbLibClient, DefaultCLITimeout)
```
Comment (sbodagala) [inline fdbclient/admin_client.go:495]: Why "Default", instead of "Max", here?

File: fdbclient/common.go [hunk 2]
```diff
@@ -45,6 +46,31 @@ const (
...
+
+		// If the status is incomplete because of a timeout, return an error. This will force a new reconciliation.
+		for _, message := range status.Cluster.Messages {
+			if message.Name == "status_incomplete_timeout" {
+				return nil, fdbv1beta2.TimeoutError{Err: fmt.Errorf("found \"status_incomplete_timeout\" in cluster messages")}
+			}
+		}
+	}
+
+	return status, nil
```
Comment (sbodagala) [inline fdbclient/common.go:71]: What about the case where the status doesn't list any processes? Are you assuming  that "messages" would be populated in that case?

**Review by sbodagala — APPROVED** [review]

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [20246c7](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/20246c7f62e2b938413f897d3f03be49e893de3d)
> * Duration 4:07:14
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/8b3d53de-c52f-46b5-b033-b02eb04072bc.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/8b3d53de-c52f-46b5-b033-b02eb04072bc/workspace.zip) zip file of the working directory (available for 30 days)

`test_operator_upgrades.log` have timed out 🤔

File: fdbclient/admin_client.go (hunk 1 above)
Comment (johscheuer) [reply fdbclient/admin_client.go:495]: Hm, that's a good call out. I can change that.

File: fdbclient/common.go (hunk 2 above)
Comment (johscheuer) [reply fdbclient/common.go:71]: Makes sense to add this case too.

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [20246c7](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/20246c7f62e2b938413f897d3f03be49e893de3d)
> * Duration 3:39:54
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/beec6258-9d09-49a5-8a88-6b9bb7a9b55a.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/beec6258-9d09-49a5-8a88-6b9bb7a9b55a/workspace.zip) zip file of the working directory (available for 30 days)

```text
• [FAILED] [124.052 seconds]
Operator when replacing a Pod stuck in Terminating state [It] should replace the Pod stuck in Terminating state [e2e, pr]
/codebuild/output/src601627202/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator/operator_test.go:800

  [FAILED] Expected
      <int>: 19
  to equal
      <int>: 20
  In [It] at: /codebuild/output/src601627202/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator/operator_test.go:807 @ 09/19/23 05:44:18.295
------------------------------
```

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [84ce2e5](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/84ce2e5a8c1ea92af10a9fcde73ee0a446f9f8ed)
> * Duration 4:07:05
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/dcfa223e-b2f8-4401-b0fb-6e34aa0e7619.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/dcfa223e-b2f8-4401-b0fb-6e34aa0e7619/workspace.zip) zip file of the working directory (available for 30 days)

I'm going to debug those in a different PR since those are failing in other PRs too, this is probably not related.

**Pull request merged**